scripts/o3de/o3de/packaging.py

```diff
@@ -0,0 +1,318 @@
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+import os
+import platform
+import argparse
+import json
+import logging
+import pathlib
+import sys
+import subprocess
+import shutil
+import pathlib
+import importlib.util
+from o3de import manifest, validation, utils, repo
+
+logger = logging.getLogger('o3de.print_registration')
+logging.basicConfig(format=utils.LOG_FORMAT)
+
+SUCCESS = 0
+
+#goal: simple CLI to package and build/automate what we need. This is proven with the prototype, so we're good to start RFC.
+# we will need sufficient time to flesh out the RFC, then the engineering plan. Otherwise it fees.
+# thankfully, I'm not blocking anyone since this is an automation task
+# can I ship this before RFC is accepted (feature flags? or separate branch) talk with Joe if this is acceptable for smallest MVP
+# all this being said, it is better that I do not overengineer at  this point. Better to write the RFC/design, get feedback, and adjust accordingly. Make sure to supply as many relevant considerations as possible
+
+#design review - ask joe about this, or if he still wants me to write RFC first. (do the minimum important things. Let the design sheperd guide you)
+
+#review automated test code for running executables
+#issue with timeouts: this can be extremely difficult
+#if someone clones a mature project...
+# SPIKE: figure out good system for timeouts. It would suck if it stalls out, but it also sucks if it prematurely times out
+# how do we show progress?
+# how do we handle logging?
+
+#add parameters for if we want to expose logging or not? otherwise render progress?
+def process_command(cmd_args, project_path):
```

Comment:
there are a lot of existing wrappers for running commands on the command line in the lytesttools folders. We should re-use an existing one if possible. I think at least one wraps subprocess.popen.